components/omega/doc/design/Tracers.md

```diff
@@ -1,415 +1,409 @@
+);
 ```
 
-#### 4.2.11 Managing time levels
+#### 4.2.9 Managing Time Levels
+
+The exact process for managing time integration is not yet clear. We
+assume a time integration module will assign and swap time indices (e.g.,
+CurTime, PrevTime) to point to the appropriate array index in all
+prognostic arrays like tracers. The current time is represented by
+an integer value of "0", while the previous time is represented as "-1",
+and so on.
 
-It is not clear yet how time integration will be managed. We assume
-here that a time integration module will be assigning and swapping
-time indices (eg CurTime, NewTime, OldTime?, etc.) to point to
-appropriate array index in all prognostic arrays like tracers.
-If these need to be managed within the tracer module, we may need
-a function like swapTimeLevels to identify and modify time level
-indices.
+#### 4.2.10 `clear`
 
-#### 4.2.12 clear
+This function clears all defined tracers and cleans up memory. It returns
+an error code if it is unable to deallocate or destroy all memory.
 
-A function is needed to clear all defined tracers and clean up
-memory. An error code is returned if it is unable to deallocate
-or destroy all memory.
 ```c++
+// Clears all defined tracers and deallocates memory.
 static int clear();
 ```
 
+Here is the corrected version with proper grammar:
+
+---
+
 ## 5 Verification and Testing
 
 ### 5.1 Test All
 
-A sample TracerDefs.inc file will be created in the test directory
-with the base tracers (T,S) and two tracers in a Debug group.
-We will assume two time levels (Cur, New). A unit test driver
-will read an input configuration that requests these two
-tracer groups. The tracers will be initialized with artificial
-data and basic arithmetic will be performed on the device. In one
-loop nest, we will perform arithmetic on one group using the
-vector of group tracer indices. In a second loop nest, we will
-perform operations on the second group using the isMembers function
-to select.
-  - tests requirements 2.1-2.4, 2.6
-
-All other requirements will need to be tested in later system
-tests once IOStreams and time integration schemes have been
-implemented.
+A sample `TracerDefs.inc` file will be located in the source directory,
+containing the base tracers and three tracers in a Debug group. We will
+assume two time levels (Cur: 0, Prev: -1). A unit test driver will read an
+input configuration that requests these two tracer groups. The tracers will
+be initialized with artificial data, and basic arithmetic will be performed
+on the device. In one loop, we will perform arithmetic on one group using
+the vector of group tracer indices. In a second loop, we will perform
+operations on the second group using the `isMembers` function to select tracers.
+
+This test covers requirements 2.1–2.4 and 2.6.
+
+All other requirements will need to be tested in later system tests once
+I/O streams and time integration schemes have been implemented.
```

Comment:
Should we be testing the validation of tracer fields using the min/max valid values or will this come later?

Reply:
@philipwjones, do you have any thoughts on this?

Reply:
It doesn't necessarily have to be a part of this PR, but it might be good to have a function like validate and/or validateAll to check tracer values for obvious errors like out-of-range. We would need a similar function in State. It might need to be up to the time integrator and/or a run-time variable to determine how frequently to check since it would impact performance. So maybe a later PR where we can decide how/when to call it?